### PR TITLE
Reorg docs

### PR DESCRIPTION
**Pull request by dagardner-nv**
* Move most of the content of `README.md` to `docs/source/getting_started.md`
* `README.md` now directs readers to other sources of documentation.
* `CONTRIBUTING.md` now included in doc builds
* `guides.rst` is now a markdown file to ensure its viewable in github
* Titles in developer guides updated to remove the prefixed numbers (hold-over from when these were all in one document)
* code of conduct & license appear in table of contents
* `docs/source/index.rst` contains links to getting started docs
* Known issues file refers to open github issues.
* Remove smart quote chars
* Fix misc parsing errors/warnings
* Myst doesn't like 'yml' as a syntax alias for 'yaml'
* Fix links to images for developer examples

relates to issue #495

### CODE REVIEW

File: morpheus/_lib/src/python_modules/common.cpp [hunk 1]
```diff
@@ -34,7 +34,6 @@ PYBIND11_MODULE(common, m)
         .. currentmodule:: morpheus.common
         .. autosummary::
            :toctree: _generate
-            TODO(Documentation)
```
Comment (dagardner-nv) [inline morpheus/_lib/src/python_modules/common.cpp:37]: This was causing a sphinx build error.

File: morpheus/pipeline/linear_pipeline.py [hunk 2]
```diff
@@ -113,23 +113,23 @@ def add_segment_boundary(self, data_type=None, as_shared_pointer=False):
...
+        >>> pipe.set_source(FileSourceStage(config, filename=val_file_name, iterative=False))
+        >>>
+        >>> # Add a segment boundary
+        >>> # [Current Segment] - [Egress Boundary] ---- [Ingress Boundary] - [Next Segment]
+        >>> pipe.add_segment_boundary(MessageMeta)
+        >>>
+        >>> # Add a sink in Segment #2
+        >>> pipe.add_stage(WriteToFileStage(config, filename=out_file, overwrite=False))
+        >>>
+        >>> pipe.run()
```
Comment (dagardner-nv) [inline morpheus/pipeline/linear_pipeline.py:132]: Unfortunately Sphinx wants the section to be named "Examples" (not singular) and the code snippets need to have the interpreters ">>>" prompts.

**Review by drobison00 — APPROVED** [review]
LGTM

File: morpheus/pipeline/linear_pipeline.py (hunk 2 above)
Comment (mdemoret-nv) [reply morpheus/pipeline/linear_pipeline.py:132]: I think this is fine. Just need to be consistent moving forwards

**Review by mdemoret-nv — APPROVED** [review]
LGTM

**Comment by dagardner-nv** [comment]
@gpucibot merge

**Pull request merged**